code/edge/src/androidTest/java/com/adobe/marketing/mobile/CompletionHandlerFunctionalTests.java

```diff
@@ -241,4 +241,47 @@ public void testSendEvent_withCompletionHandler_whenServerErrorAndHandle_callsCo
 		assertNetworkRequestCount();
 		assertTrue("Timeout waiting for EdgeCallback completion handler.", latch.await(1, TimeUnit.SECONDS));
 	}
+
+	@Test
+	public void testSendEvent_withCompletionHandler_whenExceptionThrownFromCallback_callsCompletionCorrectly()
```

Comment:
see my comment about catching the callback exception, I would expect this test to call onComplete but throw the external extension